util.go

```diff
@@ -25,6 +25,10 @@ func consistentJoin(elems []string) string {
 	return strings.Join(out, "/")
 }
 
+func consistentTrimPrefix(key, prefix string) string {
+	return strings.TrimPrefix(key, prefix+"/")
```

Comment:
Should this trim trailing slashes from `prefix` before appending a trailing slash to it to avoid double slashes in the prefix? I've had some ambiguity before where I was specifying the prefix and wasn't certain if the prefix was supposed to include the trailing slash or not.

Reply:
I think this is ok to leave, we can always change it later but this is only used internally.